### PR TITLE
Don't depend on async-std

### PR DESCRIPTION
**Pull request by dvc94ch**
before:
```
cargo tree --no-default-features --features smol_socket --edges no-dev | wc -l
169
```
after:
```
cargo tree --no-default-features --features smol_socket --edges no-dev | wc -l
135
```

### CODE REVIEW

**Comment by little-dude** [comment]
Hello,

Thanks for the PR. I haven't been following the latest async ecosystem evolutions but I assume `async-std` has been split into multiple crates, and `async-global-allocator` is one of them?

**Comment by dvc94ch** [comment]
It's nothing new, it happened when async-std adopted async-io. for most libraries I'd recommend using async-io and async-global-executor. In fact I've never needed anything else.

**Comment by dvc94ch** [comment]
just in case it wasn't clear, yes async-std uses async-global-executor to spawn tasks.

**Comment by JohnTitor** [comment]
Why did you add 8cd6a4b98db56599c7dc5145e9e7202488ce71a8? The `patch.crates-io` section on root Cargo.toml should be sufficient.

**Comment by dvc94ch** [comment]
hadn't seen that. but I haven't seen patch.crates-io used like that before. the normal way do depend on a crate in the same repo is setting the path. I noticed the paths were missing when I patched rtnetlink in my crate. if the paths aren't set correctly it fails to build because the crates are not actually compatible with the ones from crates.io.

**Comment by dvc94ch** [comment]
`[patch.crates-io]` is useful when referring to code outside of the repo.

**Comment by JohnTitor** [comment]
> hadn't seen that. but I haven't seen patch.crates-io used like that before. the normal way do depend on a crate in the same repo is setting the path. 

It's _your_ normal way, actually [actix-web do the same way](https://github.com/actix/actix-web/blob/2b2de298001c7718952a7e8647f6b1a2736c7259/Cargo.toml#L133-L143).
Setting paths to each netlink crate is annoying, and we don't have to care about it when using `patch.crates-io`. This repo contains many crates so it makes much sense here.
I would NOT accept these commits, please drop them.
(Moreover, this PR is for async-std and any unrelated changes should go to another PR.)

**Comment by little-dude** [comment]
> It's nothing new, it happened when async-std adopted async-io. for most libraries I'd recommend using async-io and async-global-executor. In fact I've never needed anything else.

Thanks!

As @JohnTitor said the `patch.crates-io` trick was done on purpose so we'd prefer not to change this.

File: rtnetlink/src/lib.rs [hunk 1]
```diff
@@ -40,6 +40,7 @@ pub mod constants;
 pub use netlink_packet_route as packet;
 pub mod proto {
     pub use netlink_proto::{Connection, ConnectionHandle, Error, ErrorKind};
+    pub use netlink_proto::packet::NetlinkMessage;
```
Comment (little-dude) [inline rtnetlink/src/lib.rs]: I'm not against adding re-exports but could this be done in a separate PR?

**Comment by dvc94ch** [comment]
Well looks like actix is your crate. It's not my way, I explained that crates fail to build with this method as the patch.crates-io is not for declaring dependencies. Here's an example: https://github.com/rust-lang/rust/blob/master/library/std/Cargo.toml

**Comment by little-dude** [comment]
> Well looks like actix is your crate. It's not my way, I explained that crates fail to build with this method as the patch.crates-io is not for declaring dependencies. Here's an example: [rust-lang/rust@`master`/library/std/Cargo.toml](https://github.com/rust-lang/rust/blob/master/library/std/Cargo.toml?rgh-link-date=2021-12-29T11%3A28%3A32Z)

I personally don't have a strong opinion on this but since this particular change is controversial, I suggest backing it out from this PR, and opening an issue where we can discuss the pros and cons of this approach. For reference here is the PR where it was implemented: https://github.com/little-dude/netlink/pull/169

**Comment by little-dude** [comment]
>  I noticed the paths were missing when I patched rtnetlink in my crate.

So is this a blocking issue for you? If so, would making a release unblock you for now?

**Comment by dvc94ch** [comment]
Like your strategy. Yes, getting a release would make this argument uninteresting enough for me to move on. However the next contributor might run into the same issue. Removed the commits and opened a new PR.

**Comment by little-dude** [comment]
> Like your strategy.

I just want you to have a short term solution while we're agreeing on a longer term solution :)

> Yes, getting a release would make this argument uninteresting enough for me to move on.

Cool, I'm a work, I'll make a release tonight if you can wait for a couple hours.

> However the next contributor might run into the same issue.

This seems problematic indeed. 

> Removed the commits and opened a new PR.

Thank you!

**Comment by dvc94ch** [comment]
The same problem happens if you use `git = "https://github.com/little-dude/netlink"`

**Comment by little-dude** [comment]
New release is available @dvc94ch : https://github.com/little-dude/netlink/releases/tag/20211229

**Comment by dvc94ch** [comment]
Thanks
